src/runtime_src/core/common/shim/buffer_handle.h

```diff
@@ -77,6 +77,20 @@ class buffer_handle
   {
     return XRT_NULL_BO;
   }
+
+  // Indicates to SHIM/driver that bh is going to be used by this BO. With
+  // offset and zie, it can also support using portion of bh (sub-BO).
+  // For now, this is only used when set_arg() is called upon an exec buf
+  // BO where pos is the arg index.
+  virtual void
+  bind_at(size_t pos, buffer_handle* bh, size_t offset, size_t size)
```

Comment:
```suggestion
  bind_at(size_t pos, const buffer_handle* bh, size_t offset, size_t size) const
```
I think const will work in these two places.  At a minimum the bh argument should be const because you will not be modifying the buffer handle or what it points to, right?

Reply:
I might have a std::map to keep track of the BO and index. I will need to modify them. bh can be const, sure.